client/src/panels/panel_declarations/finances/sobj/spend_by_so_hist.js

```diff
@@ -87,6 +95,66 @@ class SobjLine extends React.Component {
         .value(),
     };
 
+    const get_line_graph = (() => {
+      if (_.isEmpty(spending_data) && _.isEmpty(raw_data)) {
```

Comment:
Hmm, how about having one variable as Nivo props so that you can get rid of redundant props passing and if statements? This would go for same on other files

```
const is_data_empty = _.isEmpty(spending_data) && _.isEmpty(raw_data);
const empty_data_nivo_props = is_data_empty && { // props that are only for empty data
data: ...,
raw_data: ....
enableDots: ...,
},
const nivo_props = { // props for both regular data and empty data
data: ...,
raw_data:...,
margin:...,
graph_height:...,
...empty_data_nivo_props,
}
```
and then straight in render function,
`<NivoResponsiveLine {...nivo_props} />`